test/sdk/restapi/http-api-test.go.nolint

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log"
+	"os"
+	"time"
+
+	"agones.dev/agones/test/sdk/restapi/swagger"
+	"golang.org/x/net/context"
+)
+
+func main() {
+	log.Println("Client is starting")
+	time.Sleep(1 * time.Second)
+	conf := swagger.NewConfiguration()
+	portStr := os.Getenv("AGONES_SDK_HTTP_PORT")
+	conf.BasePath = "http://localhost:" + portStr
+	cli := swagger.NewAPIClient(conf)
+
+	ctx := context.Background()
+
+	// Wait for SDK server to start (5 seconds)
+	for c := 0 ; c < 5; c++ {
+		_, _, err := cli.SDKApi.Ready(ctx, swagger.SdkEmpty{})
+		if err == nil {
+			break
+		} else {
+			log.Printf("Could not send Ready: %v\n", err)
+		}
+		time.Sleep(1*time.Second)
+	}
+
+	c := make(chan string)
+
+	once := true
+	go func() {
+		gs, _, err := cli.SDKApi.WatchGameServer(ctx)
+		log.Printf("Watch response: %+v", gs)
+		if err != nil {
+			log.Printf("Error in WatchGameServer: %v\n", err)
+		} else {
+			if  gs.ObjectMeta != nil {
+				uid := gs.ObjectMeta.Uid
+				if once {
+					c <- uid
+					once = false
+				}
+			} else {
+				log.Printf("Could not parse GS\n")
+			}
+		}
+	}()
+
+	_, _, err := cli.SDKApi.Health(ctx, swagger.SdkEmpty{})
+	if err != nil {
+		log.Fatalf("Could not send health check: %v\n", err)
+	}
+
+	_, _, err = cli.SDKApi.Reserve(ctx, swagger.SdkDuration{"5"})
+	if err != nil {
+		log.Fatalf("Could not send Reserve: %v\n", err)
+	}
+
+	_, _, err = cli.SDKApi.Allocate(ctx, swagger.SdkEmpty{})
+	if err != nil {
+		log.Fatalf("Could not send Allocate: %v\n", err)
+	}
+
+	gs, _, err := cli.SDKApi.GetGameServer(ctx)
+	if err != nil {
+		log.Fatalf("Could not GetGameserver: %v\n", err)
+	}
+
+	creationTS := gs.ObjectMeta.CreationTimestamp
+
+	_, _, err = cli.SDKApi.SetLabel(ctx, swagger.SdkKeyValue{"creationTimestamp", creationTS})
+	if err != nil {
+		log.Fatalf("Could not SetLabel: %v\n", err)
+	}
+
+	// TODO: fix WatchGameServer() HTTP API Swagger definition and remove the following lines
```

Comment:
The go routine above writes to `c` as does this one. Can we delete this TODO and the below go routine? Or does the watch never fire?

Reply:
The watch function does not return, mentioned this in above comment, it is waiting for EOF, and  "result". Opened a ticket for that.